### PR TITLE
Create a way to delete a switch from Kilda cache

### LINKED ISSUE #1074: Create a way to delete a switch from Kilda cache
This must be implemented in Cache topology.
Cache topology must read the CommandMessage from Kafka and delete the switch from CacheBolt.

A part of #328 Create a way to delete switch

### PR DESCRIPTION
**Pull request by Settis**
Closes #1074 issue.

### CODE REVIEW

File: services/src/messaging/src/main/java/org/openkilda/messaging/info/event/SwitchState.java [hunk 1]
```diff
@@ -96,18 +99,11 @@ public String toString() {
...
-                .filter(item -> item.type.equalsIgnoreCase(state))
-                .findFirst()
-                .orElseThrow(() -> new IllegalArgumentException(String.format("Incorrect switch state: %s", state)));
-    }
-
     public boolean isActive() {
         return this == ADDED || this == ACTIVATED;
     }
 
     public boolean isInactive() {
```
Comment (surabujin) [inline services/src/messaging/src/main/java/org/openkilda/messaging/info/event/SwitchState.java]: Dangerous combination - 2 method that is supposed to return opposite values, but calculate it in different way. Can we drop one of this(`isActive`, `isInactive`) methods?

File: services/wfm/src/main/java/org/openkilda/wfm/topology/cache/CacheBolt.java [hunk 2]
```diff
@@ -257,6 +257,10 @@ private void handleSwitchEvent(SwitchInfoData sw, Tuple tuple, String correlatio
                 //emitRerouteCommands(affectedFlows, tuple, correlationId, FlowOperation.UPDATE, reason);
                 break;
 
+            case DELETED:
```
Comment (surabujin) [inline services/wfm/src/main/java/org/openkilda/wfm/topology/cache/CacheBolt.java]: I extremely don't like the idea to use `SwitchInfoData` for this operation. `SwitchInfoData` represent (at least now) hardware events produced by FL. Adding this new not hardware state mixes our "control" logic with hardware signals.

You need to create separate set of messages to control switch.

File: services/src/messaging/src/main/java/org/openkilda/messaging/info/event/SwitchState.java (hunk 1 above)
Comment (Settis) [reply services/src/messaging/src/main/java/org/openkilda/messaging/info/event/SwitchState.java]: Good point. I was also thinking about it. In fact `isInactive` method is not used in the project and can be deleted. As you can see, we have 3 states (`CHANGED`, `VALIDATING` & `CACHED`) which not belongs to active or inactive. This is not clear. In general, I can remove `isInactive` method.

File: services/wfm/src/main/java/org/openkilda/wfm/topology/cache/CacheBolt.java (hunk 2 above)
Comment (Settis) [reply services/wfm/src/main/java/org/openkilda/wfm/topology/cache/CacheBolt.java]: Make sense. I'll change it.

File: services/src/messaging/src/main/java/org/openkilda/messaging/info/event/SwitchState.java (hunk 1 above)
Comment (surabujin) [reply services/src/messaging/src/main/java/org/openkilda/messaging/info/event/SwitchState.java]: do it.

**Comment by Settis** [comment]
Well, you are right. Let me done #1102 first. After we will have skeleton for unit testing, I can proceed with this pull.

File: services/src/messaging/src/main/java/org/openkilda/messaging/command/switches/SwitchDeleteRequest.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    @JsonProperty("switch_id")
+    private String switchId;
+
+    /**
+     * Constructs a delete switch request.
+     *
+     * @param switchId switch id to delete rules from.
+     */
+    @JsonCreator
+    public SwitchDeleteRequest(@JsonProperty("switch_id") String switchId) {
```
Comment (surabujin) [inline services/src/messaging/src/main/java/org/openkilda/messaging/command/switches/SwitchDeleteRequest.java]: You can delegate "null check" to the lombok - add `@NonNull` here.

**Comment by Settis** [comment]
Sonar can check only non forked repositories. I've recreated this pull request with local branch #1215. Let's proceed with this new pull request.

File: services/src/messaging/src/main/java/org/openkilda/messaging/command/switches/SwitchDeleteRequest.java (hunk 3 above)
Comment (Settis) [reply services/src/messaging/src/main/java/org/openkilda/messaging/command/switches/SwitchDeleteRequest.java]: Done.

**Pull request closed without merging**